packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -62,11 +62,12 @@ class ${class_name}:
 
             grpc_channel_pool: An optional gRPC channel pool.
         """
-        self._initialization_lock = threading.Lock()
+        self._initialization_lock = threading.RLock()
         self._service_class = "${service_class}"
         self._grpc_channel_pool = grpc_channel_pool
         self._discovery_client = discovery_client
         self._stub: Optional[v2_measurement_service_pb2_grpc.MeasurementServiceStub] = None
+        self._measure_response: Optional[v2_measurement_service_pb2.MeasureResponse] = None
```

Comment:
This is not consistent with `self._measure_response = self._get_stub().Measure(request)`

In the v2 interface, `Measure` is a server-streaming RPC, so the stub's `Measure` method is actually a [UnaryStreamMultiCallable](https://github.com/shabbyrobe/grpc-stubs/blob/master/grpc-stubs/__init__.pyi#L674) object, which has a `__call__` method that returns a `CallIterator[TResponse]`. 

`MeasureResponse` is a gRPC message object, not an iterator, and Mypy shouuldn't let you do `for response in self._measure_response` when it's annotated as an `Optional[MeasureResponse]`.

Before you fix the type hint, please figure out why Mypy accepted this code with the incorrect type hint. Is it because we aren't running mypy on the codegen output?